tests/pykafka/test_balancedconsumer.py

```diff
@@ -116,6 +116,49 @@ def setUpClass(cls):
     def tearDownClass(cls):
         stop_cluster(cls.kafka)
 
+    def test_rebalance_callbacks(self):
+        def on_partitions_assigned(cns, partition_offsets):
+            self.assertTrue(len(partition_offsets) > 0)
+            held_ids = set([p.id for p in cns._get_held_partitions()])
+            self.assertEqual(held_ids | set(iterkeys(partition_offsets)), held_ids)
+            self.assigned_called = True
+            for id_ in iterkeys(partition_offsets):
+                partition_offsets[id_] = self.offset_reset
+            return partition_offsets
+
+        def on_partitions_revoked(cns, partition_offsets):
+            self.assertTrue(len(partition_offsets) > 0)
+            held_ids = set([p.id for p in cns._get_held_partitions()])
+            self.assertEqual(held_ids & set(iterkeys(partition_offsets)), set())
+            self.revoked_called = True
+
+        self.assigned_called = False
+        self.revoked_called = False
+        self.offset_reset = 50
+        try:
+            consumer_a = self.client.topics[self.topic_name].get_balanced_consumer(
+                b'test_consume_earliest', zookeeper_connect=self.kafka.zookeeper,
+                auto_offset_reset=OffsetType.EARLIEST,
+                partitions_assigned_callback=on_partitions_assigned,
+                partitions_revoked_callback=on_partitions_revoked
+            )
+            consumer_b = self.client.topics[self.topic_name].get_balanced_consumer(
+                b'test_consume_earliest', zookeeper_connect=self.kafka.zookeeper,
+                auto_offset_reset=OffsetType.EARLIEST
+            )
+            time.sleep(3)
```

Comment:
Just the other week I wrote a test helper for this, which should have been in its own pullreq - sorry - guess we can plug it in here after both get merged. See https://github.com/Parsely/pykafka/commit/6ef4199fff8b3778e96e6a0f75bd6772db33d4a9